### ISSUE TITLE
data race in k8s.io/heapster/sinks/hawkular unit test

### ISSUE THREAD

**Issue by piosz** [kubernetes/heapster#876]
```
I0113 14:10:02.499923   20174 driver.go:492] Initialised Hawkular Sink with parameters {test-heapster http://127.0.0.1:45010?tenant=test-heapster 0xc820236000 }
I0113 14:10:02.504568   20174 driver.go:492] Initialised Hawkular Sink with parameters {test-heapster http://127.0.0.1:49122?tenant=test-heapster&labelToTenant=projectId 0xc820236140 }
==================
WARNING: DATA RACE
Read by goroutine 37:
  k8s.io/heapster/sinks/hawkular.TestStoreTimeseries.func1()
      /var/lib/jenkins/workspace/project/src/k8s.io/heapster/sinks/hawkular/driver_test.go:267 +0x5c
  net/http.HandlerFunc.ServeHTTP()
      /tmp/workdir/go/src/net/http/server.go:1422 +0x47
  net/http/httptest.(*waitGroupHandler).ServeHTTP()
      /tmp/workdir/go/src/net/http/httptest/server.go:200 +0xfe
  net/http.serverHandler.ServeHTTP()
      /tmp/workdir/go/src/net/http/server.go:1862 +0x206
  net/http.(*conn).serve()
      /tmp/workdir/go/src/net/http/server.go:1361 +0x117c

Previous write by goroutine 34:
  k8s.io/heapster/sinks/hawkular.TestStoreTimeseries.func1()
      /var/lib/jenkins/workspace/project/src/k8s.io/heapster/sinks/hawkular/driver_test.go:267 +0x75
  net/http.HandlerFunc.ServeHTTP()
      /tmp/workdir/go/src/net/http/server.go:1422 +0x47
  net/http/httptest.(*waitGroupHandler).ServeHTTP()
      /tmp/workdir/go/src/net/http/httptest/server.go:200 +0xfe
  net/http.serverHandler.ServeHTTP()
      /tmp/workdir/go/src/net/http/server.go:1862 +0x206
  net/http.(*conn).serve()
      /tmp/workdir/go/src/net/http/server.go:1361 +0x117c

Goroutine 37 (running) created at:
  net/http.(*Server).Serve()
      /tmp/workdir/go/src/net/http/server.go:1910 +0x464

Goroutine 34 (running) created at:
  net/http.(*Server).Serve()
      /tmp/workdir/go/src/net/http/server.go:1910 +0x464
==================
==================
WARNING: DATA RACE
Read by goroutine 37:
  k8s.io/heapster/sinks/hawkular.TestStoreTimeseries.func1()
      /var/lib/jenkins/workspace/project/src/k8s.io/heapster/sinks/hawkular/driver_test.go:294 +0x859
  net/http.HandlerFunc.ServeHTTP()
      /tmp/workdir/go/src/net/http/server.go:1422 +0x47
  net/http/httptest.(*waitGroupHandler).ServeHTTP()
      /tmp/workdir/go/src/net/http/httptest/server.go:200 +0xfe
  net/http.serverHandler.ServeHTTP()
      /tmp/workdir/go/src/net/http/server.go:1862 +0x206
  net/http.(*conn).serve()
      /tmp/workdir/go/src/net/http/server.go:1361 +0x117c

Previous write by goroutine 34:
  k8s.io/heapster/sinks/hawkular.TestStoreTimeseries.func1()
      /var/lib/jenkins/workspace/project/src/k8s.io/heapster/sinks/hawkular/driver_test.go:294 +0x950
  net/http.HandlerFunc.ServeHTTP()
      /tmp/workdir/go/src/net/http/server.go:1422 +0x47
  net/http/httptest.(*waitGroupHandler).ServeHTTP()
      /tmp/workdir/go/src/net/http/httptest/server.go:200 +0xfe
  net/http.serverHandler.ServeHTTP()
      /tmp/workdir/go/src/net/http/server.go:1862 +0x206
  net/http.(*conn).serve()
      /tmp/workdir/go/src/net/http/server.go:1361 +0x117c

Goroutine 37 (running) created at:
  net/http.(*Server).Serve()
      /tmp/workdir/go/src/net/http/server.go:1910 +0x464

Goroutine 34 (running) created at:
  net/http.(*Server).Serve()
      /tmp/workdir/go/src/net/http/server.go:1910 +0x464
==================
==================
WARNING: DATA RACE
Read by goroutine 37:
  runtime.growslice()
      /tmp/workdir/go/src/runtime/slice.go:51 +0x0
  k8s.io/heapster/sinks/hawkular.TestStoreTimeseries.func1()
      /var/lib/jenkins/workspace/project/src/k8s.io/heapster/sinks/hawkular/driver_test.go:294 +0x8ba
  net/http.HandlerFunc.ServeHTTP()
      /tmp/workdir/go/src/net/http/server.go:1422 +0x47
  net/http/httptest.(*waitGroupHandler).ServeHTTP()
      /tmp/workdir/go/src/net/http/httptest/server.go:200 +0xfe
  net/http.serverHandler.ServeHTTP()
      /tmp/workdir/go/src/net/http/server.go:1862 +0x206
  net/http.(*conn).serve()
      /tmp/workdir/go/src/net/http/server.go:1361 +0x117c

Previous write by goroutine 34:
  k8s.io/heapster/sinks/hawkular.TestStoreTimeseries.func1()
      /var/lib/jenkins/workspace/project/src/k8s.io/heapster/sinks/hawkular/driver_test.go:294 +0x90b
  net/http.HandlerFunc.ServeHTTP()
      /tmp/workdir/go/src/net/http/server.go:1422 +0x47
  net/http/httptest.(*waitGroupHandler).ServeHTTP()
      /tmp/workdir/go/src/net/http/httptest/server.go:200 +0xfe
  net/http.serverHandler.ServeHTTP()
      /tmp/workdir/go/src/net/http/server.go:1862 +0x206
  net/http.(*conn).serve()
      /tmp/workdir/go/src/net/http/server.go:1361 +0x117c

Goroutine 37 (running) created at:
  net/http.(*Server).Serve()
      /tmp/workdir/go/src/net/http/server.go:1910 +0x464

Goroutine 34 (running) created at:
  net/http.(*Server).Serve()
      /tmp/workdir/go/src/net/http/server.go:1910 +0x464
==================
I0113 14:10:02.512116   20174 driver.go:492] Initialised Hawkular Sink with parameters {heapster http://127.0.0.1:44925?user=tester&pass=hidden 0xc8202f2000 }
I0113 14:10:02.514932   20174 driver.go:492] Initialised Hawkular Sink with parameters {heapster http://127.0.0.1:49281?filter=label(namespace_id:^$)&filter=label(container_name:^[/system.slice/|/user.slice].*)&filter=name(remove*) 0xc820236280 }
PASS
Found 3 data race(s)
FAIL	k8s.io/heapster/sinks/hawkular	1.051s
```

**Comment by piosz** [comment]
cc @burmanm @jimmidyson

**Comment by piosz** [comment]
The problem takes place in release-0.18 branch only

**Comment by piosz** [comment]
fixed by #877
